core/src/block_data_manager/mod.rs

```diff
@@ -240,12 +237,10 @@ impl BlockDataManager {
         self.db_manager.insert_instance_id_to_db(*my_instance_id);
     }
 
-    pub fn genesis_hash(&self) -> &H256 { &self.genesis_hash }
-
     pub fn genesis_state_root(&self) -> StateRootWithAuxInfo {
```

Comment:
I believe this is only for True Genesis, but I could be wrong. The current implementation is valid only for true genesis block.